### ISSUE TITLE
Averaging problem in losses

### ISSUE THREAD

**Issue by raphaelreinauer** [giotto-ai/giotto-deep#45]
In `_train_loop`, `_val_loop`, `_test_loop` the loss is averaged by the dataset size:
https://github.com/giotto-ai/giotto-deep/blob/2965f49a5653bc51790bc76c66e21f2e5ca3e65d/gdeep/pipeline/pipeline.py#L122
The problem is that, for example, the `CrossEntropyLoss` is already averaged over the minibatch
_The losses are averaged across observations for each minibatch. (https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html)_

Hence the losses are averaged twice.
